enhancements/authentication/pod-security-admission-autolabeling.md

```diff
@@ -0,0 +1,304 @@
+all the steps in 2.
+
+#### Label Synchronization Opting-in
+
+The Controller described in [PSa Label Synchronization Controller](#psa-label-synchronization-controller)
+works on an opt-in basis, meaning that users need to specifically sign themselves up
+if they wanted to stick with just SCCs.
+
+Namespaces that specifically want to have their PSa labels synchronized
+should be labelled by the `security.openshift.io/scc.podSecurityLabelSync`
+label with the label value set to `“true”`, on the other hand, to specifically
+request no PSa label synchronization, the namespace should set the label's
+value to "false".
+
+By default, namespaces without the synchronization label will be still
+considered for label synchronization. These namespaces are considered
+"no-opinion" and the label synchronization behavior for these may change
+in any future release.
+
+Having the "no-opinion" path helps keeping the older workloads working
+during upgrades. It also gives us space to decide whether we want the
+label synchronization be done by default or whether the users should
+specifically opt into it in the future.
+
+As a consequence of the opt-in/opt-out being dependent on `Namespace`
+labelling, only the privileged users can determine what kind of
+admission mechanism may run on namespaces, non-privileged users
+depend on the platform defaults.
+
+#### SCC to PSa Level Transformation
+
+The automatic PSa label synchronization requires us to map SCCs directly to a PSa
+level.
+
+In order to be able to map SCCs to PSa level automatically, it is necessary to
+introspect each field of such an SCC.
+
+SCC fields each tell the SCC authorizer how to either:
+- default a value of a given pod's `securityContext` field if unspecified
+- restrict the domain of values for a given pod's `securityContext` field
+
+PSa is based on a number of checks, each of which validates a certain `securityContext`
+field per given PSa restrictiveness level.
+
+As per the above, it is possible to introspect the SCC type fields and based on
+possible values and based on introspecting the PSa check functions for the given
+pod's `securityContext` field that's affected by the SCC field values, it is possible
+to create mappings to different PSa levels.
+
+Given the need for human interaction with PSa check functions when creating the
+SCC-\>PSa mapping, and the complexity of this task, each OpenShift version should
+always ever consider the "latest" PSa checks.
+
+### API Extensions
+
+A new `Namespace` label is being introduced - `security.openshift.io/scc.podSecurityLabelSync`.
+When set to value `"true"`, the such-labeled `Namespace` gets picked by the
+PSa label synchronization controller and this `Namespace` and its workloads
+will be reconciled according to the [PSa Label Synchronization Controller](#psa-label-synchronization-controller)
+section.
+
+### Risks and Mitigations
+
+Turning on the PSa to restricted level enforcing even with the namespace label
```

Comment:
only one that doesn't rely on SA permissions.  Builds for instance will need a cluster-wide punch through in the apiserver config.